solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRThreadModule.java

```diff
@@ -16,23 +16,69 @@
  */
 package org.apache.solr.ltr;
 import org.apache.solr.common.util.ExecutorUtil;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.SolrResourceLoader;
+import org.apache.solr.ltr.feature.Feature.FeatureWeight;
 import org.apache.solr.util.DefaultSolrThreadFactory;
+import org.apache.solr.util.SolrPluginUtils;
 
 import java.lang.management.ManagementFactory;
 import java.lang.management.ThreadMXBean;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.Executor;
+import java.util.concurrent.RunnableFuture;
 import java.util.concurrent.Semaphore;
 import java.util.concurrent.SynchronousQueue;
 import java.util.concurrent.TimeUnit;
 
 
 public class LTRThreadModule {
+
+  private static SolrResourceLoader solrResourceLoader = new SolrResourceLoader();
+
+  private static String CONFIG_PREFIX = "threadModule.";
+  private static String CONFIG_CLASS_NAME = CONFIG_PREFIX + "class";
+
+  public static LTRThreadModule getInstance(NamedList params) {
+    final LTRThreadModule threadModule;
+    final String className = (String)params.remove(CONFIG_CLASS_NAME);
+    if (className != null) {
+      Map<String,Object> threadModuleParams = new HashMap<String,Object>();
+      // gather the thread module params from amongst the general params
+      final Set<String> consumedKeys = new HashSet<String>();
+      final Iterator<Map.Entry<String,Object>> it = params.iterator();
+      while (it.hasNext()) {
+        final Map.Entry<String,Object> entry = it.next();
+        final String key = entry.getKey();
+        final Object val = entry.getValue();
+        if (key.startsWith(CONFIG_PREFIX)) {
+          threadModuleParams.put(key.substring(CONFIG_PREFIX.length()), val);
```

Comment:
No clients should be subclassing this thing (like they are expected to for Model/Feature/Norm), so there shouldn't be a need to make it pluggable.  You're jumping through extra hoops by faking a params object just so you can call invokeSetters.
I found the old constructor which took in the two limits params, and called validate internally to be very simple and easy to understand.